studio/Dockerfile

```
@@ -6,25 +6,38 @@
 #    docker build --target production --no-cache -t supabase/studio:latest .
 #    docker builder prune
 
-
-FROM node:14-slim as base
+FROM node:16-slim as base
 RUN apt-get update && apt-get install -y \
     python3 \
     build-essential \
     && rm -rf /var/lib/apt/lists/*
+
+FROM base as builder
 WORKDIR /usr/src/app
-# Do `npm ci` separately so we can cache `node_modules`
-# https://nodejs.org/en/docs/guides/nodejs-docker-webapp/
-COPY package*.json ./
-RUN npm clean-install && npm cache clean --force
-COPY . .
+COPY package*.json .
+COPY turbo.json .
+COPY packages packages
+COPY studio studio
+# the following command is still unsupported for npm // https://github.com/vercel/turborepo/issues/1830
+# RUN npm i -g turbo
+# RUN turbo prune --scope=studio --docker
+# RUN npm clean-install && npm cache clean --force
 ENTRYPOINT ["docker-entrypoint.sh"]
 
-FROM base as dev
+FROM builder as dev
+COPY --from=builder /usr/src/app ./app
+WORKDIR ./app
```

```suggestion
COPY --from=builder /usr/src/app ./
```
we can use the workdir inherited from builder stage here. also as a [best practice](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#workdir), use absolute path
> For clarity and reliability, you should always use absolute paths for your WORKDIR.